### DISCUSSION TITLE
WMR500 (STM32) The core is locked up

### DISCUSSION THREAD

**Discussion by Cla92240** [pcmihnea/HA_WMR500#4] [Q&A] [answered]
Hi pcmihnea !

First of all thank you very much for your work on WMR500.

I have tried to follow your steps but I have an issue.

I succeedded in soldering the "SWD" wires and extract the firmware with a ST-LinkV2 and ST32CubeProgrammer. And I made a back-up of it.

After that, as my firmware was 1490, I patched it like you suggested.

The problem is when I put the patched firmware in the WMR500, (I used also ST-LINK V2 and ST32CubeProgrammer), I have a weird message :"Warning: the core is locked up" and the WMR500 doesn't start. I tried to erase everything and to put back the original firmware but the message remains the same when the program starts.
![image](https://user-images.githubusercontent.com/105109248/168441306-c2a1dc95-e47b-41f0-9917-fcfe99a8b910.png)
![image](https://user-images.githubusercontent.com/105109248/168441361-c5127d05-4c36-4128-acc4-c53a4a57c5a1.png)

I wonder if you have any idea of the significance of that "LOCKED UP" status and the way to figure it out?

If not, would you agree to compare my backUp of the firmware with yours, to check if I made something wrong?

Thanks again,

Christophe

**Comment by pcmihnea** [comment, marked as answer]
Hi,

Good to know my project is useful for somebody else!

The core lock-up may be due to a invalid patch due to a different firmware - can you confirm that the previous instructions (before patching) where the same as mine?
Since the project should be completely open-source, I've removed in a previous commit my patched hex file, but it's still available in git. You could try that one - be sure to backup then erase all.

**Reply by pcmihnea** [reply]
Also be sure it's powered by micro-usb, and not batteries.
Try also a factory reset, as described in one of my previous steps.

**Reply by Cla92240** [reply]
Thank you for your reply,

In fact, I haven't found the files in Github, neither the original firmware, nor the patched one, so I can't compare with mine.

But there were "bl" instructions at the two adresses you mentionned, and I replaced them by a nop. But I will restart from the beginning to be sure.

(And the WMR was actually powered by USB, there were no batteries)

**Reply by pcmihnea** [reply]
Hint - if the previous commits aren't purged, you can see all revisions of files modified, even those deleted - just check my commit history https://github.com/pcmihnea/HA_WMR500/commits/main, then open the file deleted in my last commit https://github.com/pcmihnea/HA_WMR500/blob/c3e4291d31e20980626acd1cb2a8d3aa75134a3f/firmware/wmr500_1490_patched.hex.

**Reply by Cla92240** [reply]
Thank you!
I will test and I tell you

**Reply by Cla92240** [reply]
In fact, after comparing the two files, it appears that my backup was uncomplete from adresses 0x200 to 0x4000. Now it works.

Thank you, one more time, for that great job

**Reply by pcmihnea** [reply]
No problem!
